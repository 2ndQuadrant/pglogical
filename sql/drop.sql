
SELECT * FROM pglogical_regress_variables()
\gset

\c :provider_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_provider');

SELECT plugin, slot_type, active FROM pg_replication_slots;
SELECT count(*) FROM pg_stat_replication;

\c :provider1_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_provider1');

SELECT plugin, slot_type, active FROM pg_replication_slots;
SELECT count(*) FROM pg_stat_replication;

\c :orig_provider_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_orig_provider');

SELECT plugin, slot_type, active FROM pg_replication_slots;
SELECT count(*) FROM pg_stat_replication;

\c :subscriber_dsn
SELECT * FROM pglogical.drop_subscription('test_subscription');
SELECT * FROM pglogical.drop_subscription('test_subscription1');
SELECT * FROM pglogical.drop_node(node_name := 'test_subscriber');

\c :provider_dsn
SELECT * FROM pglogical.drop_subscription('test_orig_subscription');
SELECT * FROM pglogical.drop_node(node_name := 'test_provider');

\c :provider1_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_provider1');

\c :orig_provider_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_orig_provider');

SELECT plugin, slot_type, active FROM pg_replication_slots;
SELECT count(*) FROM pg_stat_replication;
