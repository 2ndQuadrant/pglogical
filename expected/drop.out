SELECT * FROM pglogical_regress_variables()
\gset
\c :provider_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_provider');
ERROR:  cannot drop node "test_provider" because it still has subscriptions associated with it
HINT:  drop the subscriptions first
SELECT plugin, slot_type, active FROM pg_replication_slots;
      plugin      | slot_type | active 
------------------+-----------+--------
 pglogical_output | logical   | t
 pglogical_output | logical   | t
 pglogical_output | logical   | t
(3 rows)

SELECT count(*) FROM pg_stat_replication;
 count 
-------
     3
(1 row)

\c :provider1_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_provider1');
ERROR:  cannot drop node "test_provider1" because one or more replication slots for the node are still active
HINT:  drop the subscriptions connected to the node first
SELECT plugin, slot_type, active FROM pg_replication_slots;
      plugin      | slot_type | active 
------------------+-----------+--------
 pglogical_output | logical   | t
 pglogical_output | logical   | t
 pglogical_output | logical   | t
(3 rows)

SELECT count(*) FROM pg_stat_replication;
 count 
-------
     3
(1 row)

\c :orig_provider_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_orig_provider');
ERROR:  cannot drop node "test_orig_provider" because one or more replication slots for the node are still active
HINT:  drop the subscriptions connected to the node first
SELECT plugin, slot_type, active FROM pg_replication_slots;
      plugin      | slot_type | active 
------------------+-----------+--------
 pglogical_output | logical   | t
 pglogical_output | logical   | t
 pglogical_output | logical   | t
(3 rows)

SELECT count(*) FROM pg_stat_replication;
 count 
-------
     3
(1 row)

\c :subscriber_dsn
SELECT * FROM pglogical.drop_subscription('test_subscription');
 drop_subscription 
-------------------
                 1
(1 row)

SELECT * FROM pglogical.drop_subscription('test_subscription1');
 drop_subscription 
-------------------
                 1
(1 row)

SELECT * FROM pglogical.drop_node(node_name := 'test_subscriber');
 drop_node 
-----------
 t
(1 row)

\c :provider_dsn
SELECT * FROM pglogical.drop_subscription('test_orig_subscription');
 drop_subscription 
-------------------
                 1
(1 row)

SELECT * FROM pglogical.drop_node(node_name := 'test_provider');
 drop_node 
-----------
 t
(1 row)

\c :provider1_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_provider1');
 drop_node 
-----------
 t
(1 row)

\c :orig_provider_dsn
SELECT * FROM pglogical.drop_node(node_name := 'test_orig_provider');
 drop_node 
-----------
 t
(1 row)

SELECT plugin, slot_type, active FROM pg_replication_slots;
 plugin | slot_type | active 
--------+-----------+--------
(0 rows)

SELECT count(*) FROM pg_stat_replication;
 count 
-------
     0
(1 row)

